subgraphs/alpaca-finance-lending/src/mappings/vault.ts

```diff
@@ -0,0 +1,405 @@
+  createWithdraw(
+    event,
+    market,
+    tryTokenResult.value,
+    event.params.from,
+    event.params.value
+  );
+
+  updateUserPosition(
+    event,
+    event.params.to,
+    market,
+    contract.balanceOf(event.params.to),
+    PositionSide.LENDER,
+    true
+  );
+
+  updateTokenSupply(
+    event,
+    market,
+    event.params.value.times(BIGINT_NEGATIVE_ONE)
+  );
+}
+
+export function handleAddDebt(event: Work): void {
+  const market = getMarket(event.address);
+  updateInterest(event, market);
+  const contract = Vault.bind(event.address);
+  const trytoken = contract.try_token();
+  if (trytoken.reverted) {
+    return;
+  }
+  const tryPositions = contract.try_positions(event.params.id);
+  if (tryPositions.reverted) {
+    return;
+  }
+  createBorrow(
+    event,
+    market,
+    trytoken.value,
+    tryPositions.value.getOwner(),
+    event.params.loan
+  );
+  changeMarketBorrowBalance(event, market, event.params.loan);
+  updateUserPosition(
+    event,
+    tryPositions.value.getOwner(),
+    market,
+    event.params.loan,
+    PositionSide.BORROWER,
+    false
+  );
+}
+
+export function handleRemoveDebt(event: Work): void {
+  const market = getMarket(event.address);
+  updateInterest(event, market);
+  const contract = Vault.bind(event.address);
+  const trytoken = contract.try_token();
+  if (trytoken.reverted) {
+    return;
+  }
+  const tryPositions = contract.try_positions(event.params.id);
+  if (tryPositions.reverted) {
+    return;
+  }
+  createRepay(
+    event,
+    market,
+    trytoken.value,
+    tryPositions.value.getOwner(),
+    event.params.loan
+  );
+  changeMarketBorrowBalance(
+    event,
+    market,
+    event.params.loan.times(BIGINT_NEGATIVE_ONE)
+  );
+  updateUserPosition(
+    event,
+    tryPositions.value.getOwner(),
+    market,
+    event.params.loan,
+    PositionSide.BORROWER,
+    false
+  );
+}
+
+export function handleKill(event: Kill): void {
+  const market = getMarket(event.address);
+  updateInterest(event, market);
+  const vaultContract = Vault.bind(event.address);
+  const tryConfig = vaultContract.try_config();
+  let protocolSideProfitRatio = BIGDECIMAL_ONE;
+  if (!tryConfig.reverted) {
+    const configContract = ConfigurableInterestVaultConfig.bind(
+      tryConfig.value
+    );
+    const tryGetKillBps = configContract.try_getKillBps();
+    const tryGetKillTreasuryBps = configContract.try_getKillTreasuryBps();
+    if (!tryGetKillBps.reverted && !tryGetKillTreasuryBps.reverted) {
+      protocolSideProfitRatio = tryGetKillTreasuryBps.value.divDecimal(
+        tryGetKillTreasuryBps.value.plus(tryGetKillBps.value).toBigDecimal()
+      );
+    }
+  }
+  createLiquidate(
+    event,
+    market,
+    Address.fromString(market.inputToken),
+    event.params.posVal,
+    Address.fromString(market.inputToken),
+    event.params.prize,
+    bigDecimalToBigInt(
+      event.params.prize.toBigDecimal().times(protocolSideProfitRatio)
+    ),
+    event.params.killer,
```

Comment:
Also, alpaca finance runs its own liquidation bots. What they make from them is used to buyback and burn alpaca. Should we count liquidations from those addresses entirely as protocol revenue?

Reply:
My original idea is to follow other subgraph's practice to just add the profits explicitly allocated to treasury as protocol side revenue. As said here https://docs.alpacafinance.org/tokenomics/proof-of-burn, there may be cases 3rd parties liquidation bots. Nevertheless, as right now the team are running the bots, we can count all profit as protocol side profit. Changed in 3https://github.com/messari/subgraphs/commit/71cba71400fa59bd478faf78c49fc2e8fd8708a6 Thanks!

Reply:
Oh no, I meant the opposite 😅 Sorry if I wasn't clear.

What I meant was that **some** of the liquidators are owned by alpaca, not all of them. So for those liquidations that come from those bots, 100% of the prize would be protocolSideRevenue. Whereas for the rest of the bots, only the protocol liquidation fee.

Reply:
Sorry. Revert the commit and then set default protocolSideProfitRatio in liquidation to 80% which is in line with protocol config in 7a9b54c702bb4c6b5184aed9a455ec312a24a6c9 Thanks!